src/ceph-daemon

```diff
@@ -0,0 +1,1289 @@
+                 entrypoint,
+                 args=[],
+                 volume_mounts={},
+                 cname='',
+                 podman_args=[]):
+        self.image = image
+        self.entrypoint = entrypoint
+        self.args = args
+        self.volume_mounts = volume_mounts
+        self.cname = cname
+        self.podman_args = podman_args
+
+    def run_cmd(self):
+        vols = sum(
+            [['-v', f'{host_dir}:{container_dir}']
+             for host_dir, container_dir in self.volume_mounts.items()], [])
+        envs = [
+            '-e', f'CONTAINER_IMAGE={self.image}',
+            '-e', f'NODE_NAME={get_hostname()}',
+        ]
+        cname = ['--name', self.cname] if self.cname else []
+        return [
+            podman_path,
+            'run',
+            '--rm',
+            '--net=host',
+        ] + self.podman_args + cname + envs + vols + [
+            '--entrypoint', self.entrypoint,
+            self.image
+        ] + self.args
+
+    def shell_cmd(self):
+        vols = sum(
+            [['-v', f'{host_dir}:{container_dir}']
+             for host_dir, container_dir in self.volume_mounts.items()], [])
+        envs = [
+            '-e', f'CONTAINER_IMAGE={self.image}',
+            '-e', f'NODE_NAME={get_hostname()}',
+        ]
+        return [
+            podman_path,
+            'run',
+            '-it',
+            '--net=host',
+            '--privileged',
+        ] + self.podman_args + envs + vols + [
+            '--entrypoint', '/bin/bash',
+            self.image
+        ]
+
+    def exec_cmd(self, cmd):
+        return [
+            podman_path,
+            'exec',
+            '-it',
+            self.cname,
+        ] + cmd
+
+    def run(self):
+        logging.debug(self.run_cmd())
+        return subprocess.check_output(self.run_cmd())
+
+##################################
+
+def command_version():
+    out = CephContainer(args.image, 'ceph', ['--version']).run()
+    print(out.decode('utf-8'), end='')
+    return 0
+
+##################################
+
+def command_bootstrap():
+    fsid = args.fsid or make_fsid()
+    mon_id = args.mon_id or get_hostname()
+    mgr_id = args.mgr_id or get_hostname()
+    logging.info('Cluster fsid: %s' % fsid)
+
+    logging.info('Extracting ceph user uid/gid from container image...')
+    (uid, gid) = extract_uid_gid()
+
+    # create some initial keys
+    logging.info('Creating initial keys...')
+    mon_key = CephContainer(
+        image=args.image,
+        entrypoint='/usr/bin/ceph-authtool',
+        args=['--gen-print-key'],
+    ).run().decode('utf-8').strip()
+    admin_key = CephContainer(
+        image=args.image,
+        entrypoint='/usr/bin/ceph-authtool',
+        args=['--gen-print-key'],
+    ).run().decode('utf-8').strip()
+    mgr_key = CephContainer(
+        image=args.image,
+        entrypoint='/usr/bin/ceph-authtool',
+        args=['--gen-print-key'],
+    ).run().decode('utf-8').strip()
```

Comment:
if this is used all over, why can't the run() method do this decoding and striping?

Reply:
done